### PR TITLE
Button: updating styling hooks for derivative Buttons to abstract depending on button styling

### PR DESCRIPTION
**Pull request by dzearing**
This PR makes `useToggleButtonClasses` call `useButtonClasses`. That way partners don't need to implicitly know to call the base hook.

Same applies to `useMenuButtonClasses` and `useCompoundButtonClasses`.

### CODE REVIEW

**Review by Ridvanovskyy — APPROVED** [review]

File: change/office-ui-fabric-react-2020-10-06-08-02-45-feat-variants-cherrypick.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
```
Comment (xugao) [inline change/office-ui-fabric-react-2020-10-06-08-02-45-feat-variants-cherrypick.json:2]: unexpected change file. remove

Comment (xugao) [reply change/office-ui-fabric-react-2020-10-06-08-02-45-feat-variants-cherrypick.json:2]: most of the change files look wrong; could you remove all and re-generate with accurate comment

**Review by xugao — COMMENTED** [review]
can we also remove `withThemeProvider` usage in all the button tests?

File: change/office-ui-fabric-react-2020-10-06-08-02-45-feat-variants-cherrypick.json (hunk 1 above)
Comment (dzearing) [reply change/office-ui-fabric-react-2020-10-06-08-02-45-feat-variants-cherrypick.json:2]: will do, good catch.

**Review by xugao — APPROVED** [review]

**Pull request merged**